### PR TITLE
Add minimum_licenses in the config table

### PR DESCRIPTION
**Pull request by zsinx6**
#### What
Add minimum_licenses in the config table in the features column.

#### Why
To be able to configure the minimum value using the database.

### CODE REVIEW

File: agent/lm_agent/reconciliation.py [hunk 1]
```diff
@@ -129,20 +133,26 @@ async def reconcile():
...
         product_feature = license_data["product_feature"]
         bookings_sum = license_data["bookings_sum"]
         license_total = license_data["license_total"]
         license_used = license_data["license_used"]
         slurm_used = await get_tokens_for_license(product_feature + "@flexlm", "Used")
+        config_id = await get_config_id_from_backend(product_feature)
+        minimum_value = 0
+        for config in configs:
+            if config.id == config_id:
+                minimum_value = config.features[product_feature.split(".")[1]]
```
Comment (dusktreader) [inline agent/lm_agent/reconciliation.py:148]: should there be a `break` here?

File: backend/lm_backend/api/config.py [hunk 2]
```diff
@@ -32,7 +37,22 @@ async def get_configuration(config_id: int):
             status_code=status.HTTP_404_NOT_FOUND,
             detail=(f"Couldn't get config {config_id}, " "that ID does not exist in the database"),
         )
-    return ConfigurationRow.parse_obj(fetched)
+    config_row = ConfigurationRow.parse_obj(fetched)
+    return ConfigurationItem(**config_row.dict(exclude={"features"}, features=config_row.features))
+
+
+async def get_config_id_for_product_features(product_feature: str) -> Union[int, None]:
```
Comment (dusktreader) [inline backend/lm_backend/api/config.py:46]: Need a docstring here.

File: backend/lm_backend/api/config.py [hunk 3]
```diff
@@ -32,7 +37,22 @@ async def get_configuration(config_id: int):
...
+    product, _ = product_feature.split(".")
+    query = config_table.select().where(config_table.c.product == product)
+    fetched = await database.fetch_one(query)
+    config_row = ConfigurationRow.parse_obj(fetched)
+    return config_row.id
+
+
+@router.get("/", response_model=int)
+async def get_config_id(product_feature: str):
```
Comment (dusktreader) [inline backend/lm_backend/api/config.py:56]: Also a docstring here.

File: agent/lm_agent/backend_utils.py [hunk 4]
```diff
@@ -86,6 +87,12 @@ async def get_bookings_from_backend(cluster_name: Optional[str] = None) -> List[
     return bookings
 
 
+async def get_config_id_from_backend(product_feature: str) -> int:
```
Comment (dusktreader) [inline agent/lm_agent/backend_utils.py:90]: Need a docstring here.

**Review by dusktreader — CHANGES_REQUESTED** [review]

**Review by dusktreader — APPROVED** [review]
Nice work!

File: backend/lm_backend/api/config.py [hunk 5]
```diff
@@ -1,26 +1,31 @@
...
 async def get_all_configurations():
     """
     Query database for all configurations.
     """
     query = config_table.select()
     fetched = await database.fetch_all(query)
-    return [ConfigurationRow.parse_obj(x) for x in fetched]
+    config_rows = [ConfigurationRow.parse_obj(x) for x in fetched]
+    return [
+        ConfigurationItem(**item.dict(exclude={"features"}), features=literal_eval(item.features))
```
Comment (jamesbeedy) [inline backend/lm_backend/api/config.py:23]: 👍

File: backend/lm_backend/api/config.py [hunk 6]
```diff
@@ -32,7 +37,26 @@ async def get_configuration(config_id: int):
             status_code=status.HTTP_404_NOT_FOUND,
             detail=(f"Couldn't get config {config_id}, " "that ID does not exist in the database"),
         )
-    return ConfigurationRow.parse_obj(fetched)
+    config_row = ConfigurationRow.parse_obj(fetched)
+    return ConfigurationItem(
+        **config_row.dict(exclude={"features"}), features=literal_eval(config_row.features)
```
Comment (jamesbeedy) [inline backend/lm_backend/api/config.py:42]: 👍

File: backend/lm_backend/api_schemas.py [hunk 7]
```diff
@@ -13,7 +13,23 @@ class ConfigurationRow(BaseModel):
...
+
+
+class ConfigurationItem(BaseModel):
+    """
+    A configuration parsed item.
+    """
+
+    id: Optional[int] = Field(None)
+    product: str
+    features: dict
```
Comment (jamesbeedy) [inline backend/lm_backend/api_schemas.py:32]: 👍

File: backend/lm_backend/table_schemas.py [hunk 8]
```diff
@@ -43,7 +43,7 @@
     metadata,
     Column("id", Integer, primary_key=True),
     Column("product", String),
-    Column("features", ScalarListType(str)),
+    Column("features", String),
```
Comment (jamesbeedy) [inline backend/lm_backend/table_schemas.py:46]: 👍

**Review by jamesbeedy — APPROVED** [review]
👍

**Pull request merged**